src/app/screens/sendStx/steps/Step2SelectAmount.tsx

```diff
@@ -0,0 +1,233 @@
+  getFeeForFeeRate,
+  onNext,
+  setIsLoading,
+  isLoading,
+  dustFiltered,
+  header,
+  unsignedSendStxTx,
+  setUnsignedSendStxTx,
+}: Props) {
+  const { t } = useTranslation('translation', { keyPrefix: 'SEND' });
+  const navigate = useNavigate();
+
+  const { btcFiatRate, stxBtcRate, fiatCurrency, stxBalance, stxPublicKey, feeMultipliers } =
+    useWalletSelector();
+
+  const stxToFiat = (stx: string) =>
+    getStxFiatEquivalent(
+      stxToMicrostacks(new BigNumber(stx)),
+      new BigNumber(stxBtcRate),
+      new BigNumber(btcFiatRate),
+    )
+      .toNumber()
+      .toFixed(2);
+
+  const hasStx = +stxBalance > 0;
+
+  const hasSufficientFunds = new BigNumber(stxBalance).isGreaterThanOrEqualTo(
+    new BigNumber(amount).plus(stxToMicrostacks(new BigNumber(fee ?? 0))),
+  );
+
+  useEffect(() => {
+    if (sendMax) {
+      const newAmount = new BigNumber(stxBalance).minus(stxToMicrostacks(new BigNumber(fee ?? 0)));
+
+      if (newAmount.isGreaterThan(new BigNumber(0))) {
+        setAmount(newAmount.toString());
+      }
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sendMax, fee]);
+
+  const { data: stxPendingTxData } = useStxPendingTxData();
+  const selectedNetwork = useNetworkSelector();
+
+  const [fees, setFees] = useState<FeeRates>({});
+
+  // Reactively construct a tx
+  useEffect(() => {
+    setIsLoading(true);
+
+    const createTx = async () => {
+      const rawUnsignedSendStxTx: StacksTransaction =
+        await generateUnsignedStxTokenTransferTransaction(
+          recipientAddress,
+          amount,
+          memo,
+          stxPendingTxData?.pendingTransactions ?? [],
+          stxPublicKey,
+          selectedNetwork,
+        );
+
+      // This function directly modifies unsignedSendStxTx lol
+      applyFeeMultiplier(rawUnsignedSendStxTx, feeMultipliers);
+      setUnsignedSendStxTx(buf2hex(rawUnsignedSendStxTx.serialize()));
+      setIsLoading(false);
+    };
+
+    createTx();
+  }, [
+    amount,
+    feeMultipliers,
+    memo,
+    recipientAddress,
+    selectedNetwork,
+    setIsLoading,
+    setUnsignedSendStxTx,
+    stxPendingTxData?.pendingTransactions,
+    stxPublicKey,
+  ]);
+
+  // Reactively estimate fees
+  useEffect(() => {
+    const fetchStxFees = async () => {
+      // const txRaw: string = await getRawTransaction(transaction.txid, network);
+
+      const unsignedTx: StacksTransaction = deserializeTransaction(unsignedSendStxTx);
+
+      const [low, medium, high] = await estimateTransaction(
+        unsignedTx.payload,
+        undefined,
+        selectedNetwork,
+      );
+      setFees({
+        low: Number(microstacksToStx(new BigNumber(low.fee)).toFixed(2)),
+        medium: Number(microstacksToStx(new BigNumber(medium.fee)).toFixed(2)),
+        high: Number(microstacksToStx(new BigNumber(high.fee)).toFixed(2)),
+      });
+      if (!fee)
+        setFeeRate(Number(microstacksToStx(new BigNumber(medium.fee)).toFixed(2)).toString());
```

Comment:
need to clarify with the requirements on what this logic should be. it looks to me that the notion requirements point to this section: https://www.notion.so/xverseapp/Speed-up-Stacks-transaction-Improve-fee-estimation-4455d27f8bdc4ebfbd94a16d7f18fe12?pvs=4#9509f0afd87d40eba297b357aa28763c

but there's no capping logic being applied here